creusot-contracts/src/resolve.rs

```diff
@@ -1,11 +1,17 @@
 pub use crate::base_macros::Resolve;
 use crate::*;
 
-#[trusted]
 pub trait Resolve {
     #[predicate(prophetic)]
     #[rustc_diagnostic_item = "creusot_resolve_method"]
     fn resolve(self) -> bool;
+
+    #[logic(prophetic)]
+    #[requires(structural_resolve(&self))]
+    #[ensures(self.resolve())]
+    fn resolve_coherence(self)
+    where
+        Self: Sized;
```

Comment:
We should *not* require this. If this is a problem, then let's use a shared borrow.

Reply:
this would mean changing `resolve(self) -> resolve(&self)` are you alright with that?

Reply:
We don't really have a choice. The current situation is unsound, because this means that resolve coherence is not required for unsized types.

Reply:
Yea good point. It should also be possible to only change the receiver on the coherence law, since sidedness shouldn't be enforced in pearlite